north/src/runtime/mod.rs

```diff
@@ -158,14 +158,14 @@ impl Runtime {
     }
 
     /// Installs a container in the repository
-    pub async fn install(&self, npk: &Path) -> Result<api::Response, Error> {
+    pub async fn install(&self, repo: RepositoryId, npk: &Path) -> Result<api::Response, Error> {
```

Comment:
this was exactly the reason. I would not like to use `&str` since that assumes `&RepositoryId` can coerce to a string reference but I guess that's probably a detail we can ignore for now.